test/unit/test-Lexer.js

```diff
@@ -0,0 +1,7 @@
+// If we're on a Node runtime that should support ES6, run the ES6 tests.
+if (/^v?[0-5][.]/.test(process.version)) {
```

Comment:
If one of the source files fails to load because of an error in the source when read by an ES>=6 interpreter, then `npm test` would not alert us.  `npm run-script lint` does give us some confidence here, but would not alert us if there was a failure like

```js
const A_REGEXP = new RegExp('[invalid input to RegExp constructor');
```

Putting a stake in the ground seemed the safest strategy for test code, and degrading gracefully seemed the best for production code.

Reply:
Makes sense not to just abort if the `require` fails. You can just add es6 feature detect here instead, then 👍

Reply:
Ah.  Good point.  Were you thinking of something like

```js
var es6Compatible = true;
try {
  eval('const [ x ] = [ 1234 ];');
} catch (ignored) {
  es6Compatible = false;
}
```

But I I suppose that would interact badly if tests were run under [`FLAG_disallow_code_generation_from_strings`](https://github.com/nodejs/node/pull/18212/files).

TODO(mikesamuel): Add a simple es6 canary file that can be required inside a try block.